packages/auth/src/accessToken/routes.test.ts

```diff
@@ -207,4 +208,136 @@ describe('Access Token Routes', (): void => {
       })
     })
   })
+
+  describe('Revocation', (): void => {
+    let grant: Grant
+    let token: AccessToken
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let tokenQuery: <M extends { QueryBuilderType: any }>(
+      trxOrKnex?: Objection.TransactionOrKnex | undefined
+    ) => Objection.QueryBuilderType<M>
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let actualTokenQuery: <M extends { QueryBuilderType: any }>(
+      trxOrKnex?: Objection.TransactionOrKnex | undefined
+    ) => Objection.QueryBuilderType<M>
+    let id: string
+
+    beforeAll(
+      async (): Promise<void> => {
+        actualTokenQuery = AccessToken.query
+      }
+    )
+
+    beforeEach(
+      async (): Promise<void> => {
+        grant = await Grant.query(trx).insertAndFetch({
+          ...BASE_GRANT
+        })
+        token = await AccessToken.query(trx).insertAndFetch({
+          grantId: grant.id,
+          ...BASE_TOKEN
+        })
+        tokenQuery = AccessToken.query
+        id = v4()
+      }
+    )
+
+    afterEach(
+      async (): Promise<void> => {
+        AccessToken.query = actualTokenQuery
+      }
+    )
+
+    test('Returns status 404 if token does not exist', async (): Promise<void> => {
+      const ctx = createContext(
+        {
+          headers: { Accept: 'application/json' },
+          params: { id }
+        },
+        {}
+      )
+
+      tokenQuery = () => {
+        return {
+          findById: () => {
+            return undefined
+          }
+        }
+      }
+      AccessToken.query = jest.fn(tokenQuery)
+
+      await expect(accessTokenRoutes.revoke(ctx)).rejects.toMatchObject({
+        status: 404,
+        message: 'token not found'
+      })
+    })
+
+    test('Does not modify token if it has already expired', async (): Promise<void> => {
+      const ctx = createContext(
+        {
+          headers: { Accept: 'application/json' },
+          params: { id }
+        },
+        {}
+      )
+
+      tokenQuery = () => {
+        return {
+          findById: () => {
+            return {
+              ...token,
+              expiresIn: 10000,
+              $query: () => {
+                return {
+                  patch: () => {
+                    return {
+                      where: () => Promise.resolve()
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+      AccessToken.query = jest.fn(tokenQuery)
```

Comment:
It might be better to actually insert a row into the access token table (and the grant & access tables) so that this test covers more.